src/BookReader/utils/ScrollClassAdder.js

```diff
@@ -0,0 +1,31 @@
+/** Adds a class while the given element is experiencing scrolling */
+export class ScrollClassAdder {
+  /**
+   * @param {HTMLElement} element
+   * @param {string} className
+   */
+  constructor(element, className) {
+    /** @type {HTMLElement} */
+    this.element = element;
+    /** @type {string} */
+    this.className = className;
+    this.timeout = null;
+  }
+
+  attach() {
+    this.element.addEventListener('scroll', this.onScroll);
+  }
+
+  detach() {
+    this.element.removeEventListener('scroll', this.onScroll);
+  }
+
+  onScroll = () => {
```

Comment:
Q: Do you need to bind the `onScroll` event to this class when you use this syntax? I usually have to bind methods when I add them to event listeners, ie:
```
constructor() {
  this.boundScrollHandler = this.onScroll.bind(this);
}

this.element.addEventListener('scroll', this.boundScrollHandler);
```
Otherwise they reference different instances when you remove the listener.

But I use this syntax so not sure if there's a difference.
```
onScroll() {}
```

Reply:
There is! Using `onScroll = () => { }` is a class property! Equivalent to `this.onScroll = () { ... }` in the ctor. And all `() => {}` fns have "bound" `this` vars. `onScroll() { }` is a prototype method, and follows the same rules as `Foo.prototype.onScroll = function() { ... }`, and hence needs binding. I'm not crazy over it cause it's a little subtle, but I've been using `() => {}` for anything that's going to be an event handler to avoid calling `.bind()`.